From 7b2b607ff7f68f9d5fe54477a842adad6c6a3583 Mon Sep 17 00:00:00 2001
From: Shivananda Hebbar <x0hebbar@ti.com>
Date: Fri, 19 Mar 2010 16:15:25 -0600
Subject: [PATCH 073/135] DSPBRIDGE: Change max channels/open channels type to u8

max channels can have max value of 32. So changing
max channnels and open channels type to u8

Signed-off-by: Shivananda Hebbar <x0hebbar@ti.com>
---
 arch/arm/plat-omap/include/dspbridge/_chnl_sm.h |    4 ++--
 arch/arm/plat-omap/include/dspbridge/chnlpriv.h |    6 +++---
 drivers/dsp/bridge/wmd/chnl_sm.c                |    2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
index 4cea2cc..2ef7941 100644
--- a/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
+++ b/arch/arm/plat-omap/include/dspbridge/_chnl_sm.h
@@ -125,8 +125,8 @@ struct chnl_mgr {
 	/* Critical section object handle */
 	spinlock_t chnl_mgr_lock;
 	u32 word_size;		/* Size in bytes of DSP word */
-	u32 max_channels;	/* Total number of channels */
-	u32 open_channels;	/* Total number of open channels */
+	u8 max_channels;	/* Total number of channels */
+	u8 open_channels;	/* Total number of open channels */
 	struct chnl_object **ap_channel;	/* Array of channels */
 	u8 dw_type;		/* Type of channel class library */
 	/* If no shm syms, return for CHNL_Open */
diff --git a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
index 8f5e9f4..d839eb1 100644
--- a/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
+++ b/arch/arm/plat-omap/include/dspbridge/chnlpriv.h
@@ -90,14 +90,14 @@ struct chnl_mgrinfo {
 	u8 dw_type;		/* Type of channel class library. */
 	/* Channel handle, given the channel id. */
 	struct chnl_object *chnl_obj;
-	u32 open_channels;	/* Number of open channels. */
-	u32 max_channels;	/* total # of chnls supported */
+	u8 open_channels;	/* Number of open channels. */
+	u8 max_channels;	/* total # of chnls supported */
 };
 
 /* Channel Manager Attrs: */
 struct chnl_mgrattrs {
 	/* Max number of channels this manager can use. */
-	u32 max_channels;
+	u8 max_channels;
 	u32 word_size;		/* DSP Word size. */
 };
 
diff --git a/drivers/dsp/bridge/wmd/chnl_sm.c b/drivers/dsp/bridge/wmd/chnl_sm.c
index 919a75f..a5ecc95 100644
--- a/drivers/dsp/bridge/wmd/chnl_sm.c
+++ b/drivers/dsp/bridge/wmd/chnl_sm.c
@@ -392,7 +392,7 @@ dsp_status bridge_chnl_create(OUT struct chnl_mgr **phChnlMgr,
 {
 	dsp_status status = DSP_SOK;
 	struct chnl_mgr *chnl_mgr_obj = NULL;
-	s32 max_channels;
+	u8 max_channels;
 
 	/* Check DBC requirements: */
 	DBC_REQUIRE(phChnlMgr != NULL);
-- 
1.7.0.4

